nidirect_cold_weather_payments/src/Form/ColdWeatherPaymentCheckerForm.php

```diff
@@ -184,40 +200,56 @@ public function cwpCheck(array $form, FormStateInterface $form_state) {
    */
   public function validateForm(array &$form, FormStateInterface $form_state) {
     parent::validateForm($form, $form_state);
+    // Validation is handled by ajax callback - or if no JS, by submitForm().
+  }
 
-    // Adding this validation to take care of older browsers.
-    $postcode = $form_state->getValue('postcode');
-
-    if (!empty($postcode)) {
-      $postcode_district = $this->cwpGetPostcodeDistrict($postcode);
-
-      if (!is_numeric($postcode_district) || $postcode_district < 1 || $postcode_district > 99) {
-        $form_state->setErrorByName('postcode', $this->t('Postcode must be a valid Northern Ireland postcode.'));
-      }
```

Comment:
Removing the validation logic here fixes an issue with form errors appearing at the wrong time (after refreshing the page) and in the wrong place (at the top of the page)